lang/cs/Org.Apache.REEF.Bridge.Core.Common/Client/Default/DefaultFailedRuntimeHandler.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using Org.Apache.REEF.Common.Client;
+using Org.Apache.REEF.Tang.Annotations;
+using Org.Apache.REEF.Utilities.Logging;
+using System;
+
+namespace Org.Apache.REEF.Bridge.Core.Common.Client.Default
+{
+    internal sealed class DefaultFailedRuntimeHandler : IObserver<IFailedRuntime>
+    {
+        private static readonly Logger Log = Logger.GetLogger(typeof(DefaultFailedRuntimeHandler));
+
+        [Inject]
+        private DefaultFailedRuntimeHandler()
+        {
+        }
+
+        public void OnCompleted()
+        {
+            throw new NotImplementedException();
+        }
+
+        public void OnError(Exception error)
+        {
+            throw new NotImplementedException();
+        }
+
+        public void OnNext(IFailedRuntime value)
+        {
+            Log.Log(Level.Error, $"Job {value.Id} runtime failure", value.AsError());
```

Comment:
Does value.AsError output anything?

Reply:
It outputs an exception, which then gets logged. Do you see a problem with that logic?

Reply:
Thanks for pointing out the log can take an exception -- ignore my comments on these.